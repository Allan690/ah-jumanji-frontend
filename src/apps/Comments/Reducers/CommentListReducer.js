/* eslint-disable no-case-declarations */
// Reducer for list of comments from API
import * as types from '../Actions/CommentActionTypes';

export const initialState = {
  comments: [],
};

const commentListReducer = (state=initialState, action) => {
  switch (action.type) {
    case types.FETCH_COMMENTS:
      return Object.assign(
        {},
        state,
        {comments: action.payload}
      );

    case types.FETCH_COMMENTS_FAILURE:
      return {
        ...state,
        errors: action.payload
    };

    case types.ADD_COMMENT:
      return Object.assign(
        {},
        {comments: [...state.comments, action.payload]}
      );

    case types.ADD_COMMENT_FAILURE:
      return {
      errors: action.payload
    };

    case types.DELETE_COMMENT:
      let filteredComms = state.comments.filter(comm => comm.id !== +action.payload);
      return Object.assign(
        {},
        state,
        {comments: filteredComms}
      );

    case types.EDIT_COMMENT:
      let updatedComments = state.comments.map(
        comm => (comm.id !== +action.payload.id ? comm : action.payload));
      return Object.assign(
        {},
        state,
        {comments: [...updatedComments]}
    );

    case types.EDIT_COMMENT_FAILURE:
      return {
        errors: action.payload
      };

    default:
      return state;
  }
};

export default commentListReducer;