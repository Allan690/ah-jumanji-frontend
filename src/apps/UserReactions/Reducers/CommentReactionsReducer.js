import * as types from '../Actions/CommentRxnActionsTypes';

export const initialState = {
  reactions: [],
  errors: ""
};

const commentsReactionsReducer = (state=initialState, action) => {
  switch (action.type) {
    case types.FETCH_COMMENTS_REACTIONS:
      return Object.assign(
        {},
        state,
        {reactions: action.payload}
      );

    case types.FETCH_COMMENTS_REACTIONS_FAILURE:
      return Object.assign(
        {},
        state,
        {errors: action.payload}
      );

    case types.ADD_COMMENT_REACTION:
      return Object.assign(
        {},
        state,
        {reactions: [...state.reactions, action.payload]}
      );

    case types.UNDO_COMMENT_REACTION:
      return Object.assign(
        {},
        state,
        {reactions: [state.reactions.filter(rxn => rxn.comment !== action.payload)]}
      );

    case types.ADD_COMMENT_REACTION_FAILURE:
      return Object.assign(
        {},
        state,
        {errors: action.payload}
      );

    default:
      return state;
  }
};

export default commentsReactionsReducer;