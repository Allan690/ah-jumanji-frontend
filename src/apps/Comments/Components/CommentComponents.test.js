import React from 'react';
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import configureStore from 'redux-mock-store';

import CommentComponent from "./Comment";
import {initialState} from "../Reducers/CommentListReducer";

Enzyme.configure({adapter: new Adapter()});
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const shallowWithStore = (component, store) => {
  const context = {
    store,
  };
  return shallow(component, {context});
};

const fakeComment = {
  "id": 100,
  "author": {
    "username": "dmithamo",
    "image": "",
  },
  "body": "fake body"
};

describe('<CommentComponent />', () => {
  it('render all expected bits of a commnet', () => {
    const component = shallow(<CommentComponent comment={fakeComment} />);
    expect(component).toMatchSnapshot();
  });
});

describe('<CommentComponent />', () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <CommentComponent comment={fakeComment} />
        </MemoryRouter>
      </Provider>
    );
  });

  it('renders without crashing', () => {
    shallowWithStore(<CommentComponent comment={fakeComment} />, store);
  });

  it('renders the body of the comment ', () => {
    const commBody = wrapper.find('.comment-body').first().text();
    expect(commBody).toEqual(fakeComment.body);
  });

  it('renders the prof pic of the user', () => {
    const img = wrapper.find('.img').first();
    expect(img).toBeTruthy;
  });

});