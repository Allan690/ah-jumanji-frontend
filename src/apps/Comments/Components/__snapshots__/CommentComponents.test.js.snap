// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentComponent /> render all expected bits of a commnet 1`] = `
<div>
  <li
    className="list-group-item"
    id={100}
  >
    <div
      className="row"
    >
      <div
        className="col-md-1 col-sm-2 col-2"
      >
        <img
          alt="follower avatar"
          className="followers-avatar message-avatar rounded-circle"
          src="https://cdn-images-1.medium.com/max/1200/1*MccriYX-ciBniUzRKAUsAw.png"
        />
      </div>
      <div
        className="col-md-11 col-sm-10 col-10"
      >
        <div
          className="message"
        >
          <Link
            replace={false}
            to="/ah-jumanji/profile/dmithamo"
          >
            Dmithamo
          </Link>
          <span
            className="float-right text-muted"
          >
            <i
              className="fas fa-clock"
            />
             Date undefined undefined
          </span>
          <br />
          <span
            className="comment-body"
          >
            fake body
          </span>
        </div>
      </div>
    </div>
  </li>
  <Connect(AddCommentComponent)
    actionToExecute="edit"
    specialClass="hidden"
  />
  <hr />
</div>
`;
